### PR TITLE
Added ;;1, ;;2, etc for selecting search results

### LINKED ISSUE #98: Allow selecting just by ;;1 instead of ;;play 1
Allow selecting just by ;;1 instead of ;;play 1 when presented with the choices after a search

Picked this feature request up from the #suggestions channel

### PR DESCRIPTION
**Pull request by SuhanKoh**
Fixes #98 
When given a list of track(s) to select, it allows the user to enter ";;1" or more to select the track.

### CODE REVIEW

File: FredBoat/src/main/java/fredboat/command/music/control/NumericSelectTrackCommand.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import net.dv8tion.jda.core.entities.Member;
+import net.dv8tion.jda.core.entities.Message;
+import net.dv8tion.jda.core.entities.TextChannel;
+import org.apache.commons.lang3.StringUtils;
+
+public class NumericSelectTrackCommand extends Command implements IMusicCommand, ICommandRestricted {
+
+    @Override
+    public PermissionLevel getMinimumPerms() {
+        return PermissionLevel.USER;
```
Comment (Frederikam) [inline FredBoat/src/main/java/fredboat/command/music/control/NumericSelectTrackCommand.java]: This should be placed in the bottom of the file for the sake of consistency.

File: FredBoat/src/main/java/fredboat/util/rest/SearchUtil.java [hunk 2]
```diff
@@ -39,6 +39,8 @@
 
 public class SearchUtil {
 
+    public static final int NUMBER_OF_SEARCH = 5;
```
Comment (Frederikam) [inline FredBoat/src/main/java/fredboat/util/rest/SearchUtil.java]: Something like MAX_RESULTS would be more logical

File: FredBoat/src/main/java/fredboat/commandmeta/init/MusicCommandInitializer.java [hunk 3]
```diff
@@ -67,6 +67,7 @@ public static void initCommands() {
         CommandRegistry.registerCommand("join", new JoinCommand(), "summon", "jn");
         CommandRegistry.registerCommand("leave", new LeaveCommand(), "lv");
         CommandRegistry.registerCommand("select", new SelectCommand(), "sel");
+        CommandRegistry.registerCommand("selecttrack", new NumericSelectTrackCommand(), buildNumericalSelectAllias());
```
Comment (Frederikam) [inline FredBoat/src/main/java/fredboat/commandmeta/init/MusicCommandInitializer.java]: `selecttrack` should not be registered. Don't be afraid of registering a command type more than once though.

**Comment by SuhanKoh** [comment]
- [ ] Remove NumericSelectTrackCommand.java
- [ ] Change NUMBER_OF_SEARCH to MAX_RESULTS
- [ ] Remove selecttrack
- [ ] Reuse ;;select instead

File: FredBoat/src/main/java/fredboat/commandmeta/init/MusicCommandInitializer.java [hunk 4]
```diff
@@ -131,12 +130,17 @@ public static void initCommands() {
     /**
      * Build a string array that consist of the max number of searches.
      *
-     * @return String array that contains string representation of numbers.
+     * @param addOnAliases Aliases to be appended to the rest of the ones being built.
```
[the remark this answers is not in the document]
Comment (SuhanKoh) [reply FredBoat/src/main/java/fredboat/commandmeta/init/MusicCommandInitializer.java]: Got it

**Comment by Frederikam** [comment]
One thing that bugs me is that saying something like `;;2 3` will pick the 3rd song. Ideally, this should pick the 2nd song in my opinion.

**Comment by napstr** [comment]
Is this done? I will take care of resolving the conflict when it is.

**Comment by SuhanKoh** [comment]
@napstr Yes

**Comment by Frederikam** [comment]
Sorry for the delay

**Comment by SuhanKoh** [comment]
No worries.

**Pull request merged**